query_engine/src/datafusion_impl/physical_planner.rs

```diff
@@ -81,18 +68,28 @@ impl DatafusionPhysicalPlannerImpl {
             .extensions
             .insert(ceresdb_options);
 
-        let logical_optimize_rules = Self::logical_optimize_rules();
-        let state =
+        // Using default logcial optimizer, if want to add more custom rule, using
+        // `add_optimizer_rule` to add.
+        let mut state =
             SessionState::with_config_rt(df_session_config, Arc::new(RuntimeEnv::default()))
-                .with_query_planner(Arc::new(QueryPlannerAdapter))
-                .with_analyzer_rules(Self::analyzer_rules())
-                .with_optimizer_rules(logical_optimize_rules);
+                .with_query_planner(Arc::new(QueryPlannerAdapter));
+
+        // Our analyzer has high priority, so first add we custom rules, then add the
+        // default ones.
+        state = state.with_analyzer_rules(Self::analyzer_rules());
```

Comment:
Define a function it like `register_custome_analyzer_rules` like iox and place this into it?